### PR TITLE
[CPL][APPWIZ] Attempt to enable Visual Style

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

JIRA issue: [CORE-16669](https://jira.reactos.org/browse/CORE-16669)

- Load `comctl32.dll` and call the `InitCommonControls` function to enable Visual Style.

### CODE REVIEW

**Comment by HBelusca** [comment]
I don't see why this is needed.... Ctrl-A selection must work in regular edit boxes, whether or not you have comctl32 loaded.

**Comment by learn-more** [comment]
If you run this on windows, is it also required to call this?

**Comment by katahiromz** [comment]
wont-work
It won't work on 2k3...

File: dll/cpl/appwiz/createlink.c [hunk 1]
```diff
@@ -597,11 +597,38 @@ ShowCreateShortcutWizard(HWND hwndCPl, LPWSTR szPath)
...
+    hComCtl32 = LoadLibraryW(L"comctl32.dll");
+#if 1
+    fn = (FN_InitCommonControls)GetProcAddress(hComCtl32, (LPSTR)(DWORD_PTR)15);
+#else
+    fn = (FN_InitCommonControls)GetProcAddress(hComCtl32, "InitCommonControls");
+#endif
+    if (fn)
+        (*fn)();
+    else
+        MessageBoxW(NULL, L"InitCommonControls failed.", L"appwiz.cpl", MB_ICONERROR);
```
Comment (Fraizeraust) [inline dll/cpl/appwiz/createlink.c]: I'm quite allergic to such message boxes that give no meaningful information whatsoever about the failure. Either populate the box description with a more concise detail (and make sure you don't hard-code the box description, just implement it within resources) OR print out a debug info instead.

**Comment by katahiromz** [comment]
https://docs.microsoft.com/en-us/windows/win32/controls/cookbook-overview#using-comctl32-version-6-in-control-panel-or-a-dll-that-is-run-by-rundll32exe
This article says:
```txt
1. Link to ComCtl32.lib and call InitCommonControls.
2. Add a file called YourApp.cpl.manifest to your source tree that has the XML manifest format.
3. Add the manifest to your application's resource file as resource ID 123.
```

File: dll/cpl/appwiz/createlink.c (hunk 1 above)
Comment (HBelusca) [reply dll/cpl/appwiz/createlink.c]: Also since execution can continue even if that function failed it's useless indeed to display an interruptive msgbox.

**Comment by katahiromz** [comment]
Still Ctrl+A won't work in 2k3. Something is wrong.

**Comment by Fraizeraust** [comment]
You can try testing some edit boxes from other programs/applets/whatever and see if CTRL + A gets triggered. If not then we have a bug on USER (Win32k) that prevents the said keyboard event from being consumed.

**Comment by HBelusca** [comment]
> It seems like 123 is a special value...

Very simply change that value, recompile and retest to see whether it has effects.

> You can try testing some edit boxes from other programs/applets/whatever and see if CTRL + A gets triggered. If not then we have a bug on USER (Win32k) that prevents the said keyboard event from being consumed.

Katayama also tested our wizard in Windows and reported the ctrl-A for the wizard's edit-box didn't work too, so it's not a problem in our user32/win32k.

**Comment by katahiromz** [comment]
https://jira.reactos.org/secure/attachment/55988/a.zip
This dialog program contains no manifest. This program doesn't accepts Ctrl+A in Windows.

**Comment by HBelusca** [comment]
@katahiromz : Try using `return TRUE;` instead of `break;` in the handlers of `WM_INITDIALOG` ?

**Comment by HBelusca** [comment]
Oooook..... Some people on the internet claim this:
https://stackoverflow.com/questions/291792/win32-select-all-on-edit-ctrl-textbox

**Comment by katahiromz** [comment]
> Try using return TRUE; instead of break; in the handlers of WM_INITDIALOG ?
There is no difference except focus.

**Comment by HBelusca** [comment]
Well from what I can see it may be that either we need to manually handle Ctrl-A ourselves, or, do as you wanted initially to do and load the comctl32 if available (in which case this would allow to have the themes correctly applied if e.g. the wizard is used on Windows XP or on ReactOS with theming enabled).

**Comment by katahiromz** [comment]
VirtualBox_ReactOS_08_02_2020_04_20_37
M$ `appwiz.cpl` won't work in ReactOS. `DUSER.dll` requested.

**Comment by HBelusca** [comment]
Not a problem, just add as many necessary dlls as needed.

**Comment by katahiromz** [comment]
I added 2k3 `duser.dll` into ReactOS and retried to create shortcut in ReactOS with 2k3 `appwiz.dll`.
not-working
Ctrl+A is not working yet.

not-working-2
Even in 2k3, Ctrl+A is not working.

**Comment by katahiromz** [comment]
So, the problem is manifest resource ID 123 and Activation Context?

**Comment by HBelusca** [comment]
Actually I checked other dialogs with edit controls on Windows / ReactOS and none of them support Ctrl-A as a keyboard accelerator by default.

**Pull request merged**